diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt

```diff
@@ -90,6 +93,42 @@ class HeaderCommentRule(private val configRules: List<RulesConfig>) : Rule("head
                 )
             }
         }
+
+        val copyrightWithCorrectYear = makeCopyrightCorrectYear(copyrightText)
+
+        if (copyrightWithCorrectYear != ""){
+            WRONG_COPYRIGHT_YEAR.warnAndFix(configRules, emitWarn, isFixMode, fileName, node.startOffset) {
+                (headerComment as LeafElement).replaceWithText(headerComment.text.replace(copyrightText, copyrightWithCorrectYear))
+            }
+        }
+    }
+
+    private fun makeCopyrightCorrectYear(copyrightText: String) : String {
+        var regex = Regex("\\b(\\d+-\\d+)\\b")
+        val hyphenYear = regex.find(copyrightText)
+        val curYear = LocalDate.now().year
+
+        if (hyphenYear != null) {
+            val copyrightYears = hyphenYear.value.split("-")
+            if (copyrightYears[1].toInt() != curYear) {
+                val validYears = "${copyrightYears[0]}-${curYear}"
+                return copyrightText.replace(regex, validYears)
+            }
+        }
+
+        regex = Regex("((©|\\([cC]\\))+ *\\d+)")
```

Comment:
Also you can use triple-quoted strings in regex, because they don't require special character escaping